src/System.Runtime/tests/System/Version.cs

```diff
@@ -93,14 +93,18 @@ public static IEnumerable<object[]> CompareToTestData()
         yield return new object[] { new Version(1, 2), new Version(1, 0, 1), 1 };
         yield return new object[] { new Version(1, 2), new Version(1, 0, 0, 1), 1 };
 
+        yield return new object[] { new Version(3, 2, 1), new Version(2, 2, 1), 1 };
+        yield return new object[] { new Version(3, 2, 1), new Version(3, 1, 1), 1 };
+        yield return new object[] { new Version(3, 2, 1), new Version(3, 2, 0), 1 };
+
         yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 4), 0 };
         yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 5), -1 };
         yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 3), 1 };
     }
 
     [Theory]
     [MemberData(nameof(CompareToTestData))]
-    public static void TestCompareTo(Version version1, Version obj, int expectedSign)
+    public static void TestCompareTo(Version version1, object obj, int expectedSign)
```

Comment:
Rather than change this from ```Version obj``` to ```object obj```, would it not be better to change it to ```Version version2``` and get rid of the first line in the test?